src/Cake.Curl/CurlSettings.cs

```diff
@@ -138,5 +138,25 @@ public class CurlSettings : ToolSettings
         /// This option is used together with <a href="https://curl.haxx.se/docs/manpage.html#--retry">--retry</a>.
         /// </remarks>
         public bool RetryOnConnectionRefused { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum time in seconds that you allow the whole operation to take.
+        /// </summary>
+        /// <remarks>
+        /// This is useful for preventing your batch jobs from hanging for hours due to slow networks or links going down.
+        /// Since curl version 7.32.0, this option accepts decimal values, but the actual timeout will decrease in accuracy as the specified timeout increases in decimal precision.
+        /// See also <a href="https://curl.haxx.se/docs/manpage.html#--connect-timeout">--connect-timeout</a>.
+        /// </remarks>
+        public double MaxTimeSeconds { get; set; }
```

Comment:
This seems like a good opportunity to make [curl’s syntax](
https://curl.haxx.se/docs/manpage.html#-m) a bit more user-friendly. I suggest that we expose this value as a `Nullable<TimeSpan>` instead of `double` and rename the option to `MaxTime`.

That way, we let the user decide how they want to specify a timeout value without forcing them to use decimal numbers. For example, to set a timeout of _one minute and a half_ they could use any of these formats:

```csharp
settings.MaxTime = TimeSpan.FromMinutes(1.5);
settings.MaxTime = TimeSpan.FromSeconds(90);
settings.MaxTime = new TimeSpan(hours: 0, minutes: 1, seconds: 30);
```

Using a `TimeSpan` would also be more in line with the rest of the Cake API (see [`ToolSettings.ToolTimeout`](
https://cakebuild.net/api/Cake.Core.Tooling/ToolSettings/B98A10B0)).